### PR TITLE
Make `mypy` pass from the command line 

### LINKED ISSUE #6029: [CT-1315] [Internal] Make mypy pass from the command line
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Stub issue for https://github.com/dbt-labs/dbt-core/pull/5983. Contributing.md states it's required, and only my PRs with issues have been reviewed

### Describe alternatives you've considered

_No response_

### Who will this benefit?

_No response_

### Are you interested in contributing this feature?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by max-sixty**
Similar issue to https://github.com/dbt-labs/dbt-core/pull/5947 ~(which this includes, and so requires merging #5947 first)~ — currently running `mypy` from the command line fails, because the configs are in `.pre-commit-config.yaml`, which isn't available to editors etc.

### Description

closes #6029 

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Comment by lostmygithubaccount** [comment]
closes #6029

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,9 @@
+[tool.mypy]
+exclude = ['core/build/.*']
+files = 'core'
```
Comment (emmyoop) [inline pyproject.toml:3]: This is actually less restrictive than now.  We currently restrict to `core/dbt` in `.pre-commit-config.yaml`.  If we alter this to only look there, you no longer need the exclude.

Would it be possible to only define the `files` key here and remove it from `.pre-commit-config.yaml` so we don't have it defined in multiple places for mypy?

**Review by emmyoop — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (max-sixty) [reply pyproject.toml:3]: > Would it be possible to only define the `files` key here and remove it from `.pre-commit-config.yaml` so we don't have it defined in multiple places for mypy?

Yeah, annoyingly mypy doesn't have a notion of "force exclude", where it excludes something even if explicitly passed it on the command line. And pre-commit passes file names. So it's still necessary to have the duplication unfortunately

Comment (max-sixty) [reply pyproject.toml]: > This is actually less restrictive than now. We currently restrict to `core/dbt` in `.pre-commit-config.yaml`. If we alter this to only look there, you no longer need the exclude.

Nice, done

**Review by emmyoop — APPROVED** [review]
Thanks @max-sixty!  Really appreciate the contribution here!

**Pull request merged**